e2e/execution_test.go

```diff
@@ -168,6 +261,71 @@ func testExecution(t *testing.T) {
 	t.Run("list", func(t *testing.T) {
 		resp, err := client.ExecutionClient.List(context.Background(), &pb.ListExecutionRequest{})
 		require.NoError(t, err)
-		require.Len(t, resp.Executions, 2)
+		require.Len(t, resp.Executions, 4)
+	})
+
+	t.Run("many executions in parallel", func(t *testing.T) {
+		var (
+			n          = 10
+			executions = make([]hash.Hash, 0)
+			taskKey    = "task1"
+			inputs     = &types.Struct{
+				Fields: map[string]*types.Value{
+					"msg": {
+						Kind: &types.Value_StringValue{
+							StringValue: "test",
+						},
+					},
+				},
+			}
+		)
+		t.Run("create executions", func(t *testing.T) {
+			wg := sync.WaitGroup{}
+			var mutex sync.Mutex
+			wg.Add(n)
+			for i := 1; i <= n; i++ {
+				go func(i int) {
+					defer wg.Done()
+					resp, err := client.ExecutionClient.Create(context.Background(), &pb.CreateExecutionRequest{
+						TaskKey:      taskKey,
+						EventHash:    hash.Int(1111 + i),
```

Comment:
```suggestion
						EventHash:    hash.Int(i),
```

- start i from 1000 (with comment)
- or maybe user hash.Random which will fit it here.